modxcms/revolution
# [Port to 3.x] Form customization front end fixes/improvements

### What does it do?
Reworks the methods responsible for moving and changing field/tab visibility and labelling. Code optimizations were made and method documentation was added. Note this is a port of PR #15836, which will ultimately be withdrawn as it creates breaking changes not desirable for 2.x.

### Why is it needed?
Most of these methods were written to receive/process serialized/array data, but there appears to be no reason (any longer at least) to keep them that way—especially given that problems have arisen such as the one referenced by this PR. The changes allow for the use of commas in labels and provides simplification and clarification of the relevant code.

### How to test
Clear your caches and remember to run `grunt build` within the `_build/templates/default` directory to compile the changes into the main js file. Create one or two test TVs, then create a Form Customization profile with at least one rule. Within your rule(s), experiment with re-labelling and hiding various resource fields along with your test TVs. Also create a new tab in Regions tab and verify that assigning your TVs to that tab (or the standard regions) works as expected.

### Related issue(s)/PR(s)
Resolves #15357.
